src/js/core/methods/helpers/ethereumSignTypedData.js

```diff
@@ -0,0 +1,190 @@
+/* @flow */
+
+import BigNumber from 'bignumber.js';
+import type { EthereumSignTypedDataTypes } from '../../../types/networks/ethereum';
+import type { EthereumFieldType } from '../../../types/trezor/protobuf';
+import { Enum_EthereumDataType } from '../../../types/trezor/protobuf';
+import { ERRORS } from '../../../constants';
+import { stripHexPrefix } from '../../../utils/formatUtils';
+
+// Copied from https://github.com/ethers-io/ethers.js/blob/v5.5.2/packages/abi/src.ts/fragments.ts#L249
+const paramTypeArray = new RegExp(/^(.*)\[([0-9]*)\]$/);
+const paramTypeBytes = new RegExp(/^bytes([0-9]*)$/);
+const paramTypeNumber = new RegExp(/^(u?int)([0-9]*)$/);
+
+/**
+ * Parse the given EIP-712 array type into its entries, and its length (if not dynamic)
+ * E.g. `uint16[32]` will return `{entryTypeName: 'uint16', arraySize: 32}`.
+ */
+export function parseArrayType(arrayTypeName: string): {
+    entryTypeName: string,
+    arraySize: number | null,
+} {
+    const arrayMatch = paramTypeArray.exec(arrayTypeName);
+    if (arrayMatch === null) {
+        throw ERRORS.TypedError(
+            'Method_InvalidParameter',
```

Comment:
Please change occurences `Method_InvalidParameter` to `Runtime` in this file.

Reply:
Done in https://github.com/trezor/connect/pull/983/commits/d9bbb80f2e5bc78e37272737baf9ce856bfdaf52 along with the same changes in `src/js/core/methods/EthereumSignTypedData.js`.